### PR TITLE
feat: Dev Portal SAML SSO

### PR DESCRIPTION
**Pull request by cloudjumpercat**
### Description

SAML SSO is being added to Dev Portal. This PR adds instructions for it and creates SSO includes so that org SSO and Dev Portal SSO content can be shared.

This PR also fixes some outdated Okta SSO language, see DOCU-4120 for more information.
 

DOCU-4135
### Testing instructions

Preview links: 

- [SSO attribute mapping reference](https://deploy-preview-8118--kongdocs.netlify.app/konnect/reference/sso/)
- [Konnect Generic SSO how to](https://deploy-preview-8118--kongdocs.netlify.app/konnect/org-management/sso/)
- [Dev Portal Generic SSO how to](https://deploy-preview-8118--kongdocs.netlify.app/konnect/dev-portal/access-and-approval/sso/)
- [Konnect Okta SSO how to](https://deploy-preview-8118--kongdocs.netlify.app/konnect/org-management/okta-idp/)
- [Dev Portal Okta SSO how to](https://deploy-preview-8118--kongdocs.netlify.app/konnect/dev-portal/access-and-approval/okta-idp/)

### Checklist 

- [x] Review label added 
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

### CODE REVIEW

**Comment by cloudjumpercat** [comment]
I'm still working on testing team mappings and Dev Portal SAML, but the draft docs are now ready for tech review! 
- Dev Portal Generic SSO how to and Konnect Generic SSO how to
  - In the PR, the content for these pages now lives primarily in an include since the content for both generic SSO pages is so similar. As you look over these docs, I'd appreciate any feedback if I missed anything or need to fix any instructions that relates to one (ex. Konnect org) but not the other (ex. Dev Portal) that I missed on accident.
- Konnect Okta SSO how to and Dev Portal Okta SSO how to
  - Same comment here about how we're now using an include for the content because the two sets of instructions are very similar. This just makes it easier to keep the content consistent since the updates to one will show up in the other.
- SSO attribute mapping reference
  - This content already existed, I just moved it to it's own page. I still have an open question about what the Konnect values are and where in Konnect those values map to

File: app/_includes/md/konnect/generic-sso.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+* {{site.konnect_short_name}} must be added to your IdP as an application
+* Users that need to use SSO are added to your IdP tenant
+* Claims are set up in your IdP
+
+
+## Set up SSO in {{site.konnect_short_name}}
+{% navtabs %}
+{% navtab OIDC%}
+{% if include.desc == "Konnect Org" %}
```
Comment (cloudjumpercat) [inline app/_includes/md/konnect/generic-sso.md:13]: For reviewers: This label in the include file pulls the content from the `{% if include.desc == "Konnect Org" %}` to the next `{% endif %}` into _only_ the Konnect org SSO docs, not the Dev Portal SSO docs. This lets us share most of the content and conditionally render content that is specific to one feature and not the other, like the UI instructions to get to the SSO settings in Konnect.

File: app/_includes/md/konnect/generic-sso.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+1. In [{{site.konnect_saas}}](https://cloud.konghq.com/login), click {% konnect_icon organizations %} **Organization** > **Settings**, and then click the **Authentication Scheme** tab.
+{% endif %}
+{% if include.desc == "Dev Portal" %}
```
Comment (cloudjumpercat) [inline app/_includes/md/konnect/generic-sso.md:16]: Same comment here as the previous, but everything in this `if` tag is only pulled into the Dev Portal SSO docs, not the Konnect org SSO docs.

File: app/_includes/md/konnect/okta-sso.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+1. Optional: If you want to use group claims for Konnect [developer team mappings](/konnect/dev-portal/access-and-approval/add-teams/), click the **Claims** tab in Okta to [configure a `groups` claim](https://developer.okta.com/docs/guides/customize-tokens-groups-claim/main/#add-a-groups-claim-for-a-custom-authorization-server) and fill in the following fields:
+
+    | Name    | Name format  | Filter          | Filter Value |
+    |---------|--------------|-----------------|--------------|
+    | groups  | Unspecified  | Matches regex   | .*           |
+{% endif %}
+
+1. [Add users to the Okta application](https://help.okta.com/en-us/content/topics/users-groups-profiles/usgp-assign-apps.htm).
+
+1. [Generate a signing certificate](https://help.okta.com/en-us/content/topics/apps/manage-signing-certificates.htm) to use in {{site.konnect_short_name}}.
```
Comment (cloudjumpercat) [inline app/_includes/md/konnect/okta-sso.md:71]: There's no place to put the cert in konnect, so I just need to generate it and activate it?

File: app/_includes/md/konnect/generic-sso.md (hunk 1 above)
Comment (smritikjaggi) [reply app/_includes/md/konnect/generic-sso.md:13]: Very powerful! Thank you!

File: app/_includes/md/konnect/generic-sso.md (hunk 2 above)
Comment (smritikjaggi) [reply app/_includes/md/konnect/generic-sso.md:16]: Very powerful! Thank you!

File: app/_includes/md/konnect/generic-sso.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    * **Claim Mappings**: Customize the mapping of required attributes to a different claim in the `id_token` {{site.konnect_short_name}} receives from the IdP. By default, {{site.konnect_short_name}} requires three attributes: Name, Email, and Groups. The values in these attributes are mapped as follows:
+        * `name`: Used as the {{site.konnect_short_name}} account's `full_name`.
+        * `email`: Used as the {{site.konnect_short_name}} account's `email`.
+        * `groups`: Used to map users to teams defined in the team mappings upon login.
+
+1. After clicking **Save**, close the configuration dialog and from the OIDC context menu, click **Enable OIDC**.
+{% endnavtab %}
+{% navtab SAML %}
+
+The {{site.konnect_short_name}} SAML integration allows you to configure various identity providers. While technically any SAML-compliant provider can be used, the following have been verified:
```
Comment (smritikjaggi) [inline app/_includes/md/konnect/generic-sso.md:52]: Do we want to add a similar list for OIDC as well?

File: app/_includes/md/konnect/generic-sso.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+Test the SSO configuration by navigating to the callback URL for your Dev Portal. For example: `https://{portalId}.{region}.portal.konghq.com/login`.
+{% endif %}
+
+If the configuration is correct, you will see the IdP sign-in page. 
+
+You can now manage your organization's user permissions entirely from the IdP application.
+
+## Troubleshooting
+
+<details><summary>Troubleshooting authentication issues with large numbers of groups</summary>
```
Comment (smritikjaggi) [inline app/_includes/md/konnect/generic-sso.md:106]: @Daniel2218 / @joshcman 
Do we want to add a section similar to https://docs.konghq.com/konnect/org-management/sso/#reference for Dev Portal?

File: app/konnect/dev-portal/access-and-approval/sso.md [hunk 6]
```diff
@@ -12,116 +12,9 @@ Keep the following in mind when configuring SSO for Dev Portal:
...
-| {{site.konnect_short_name}} value      | Azure value | Okta value | Auth0 value |
-| ----------- | ----------- | ----------- | ----------- |
-| Provider URL      | The value stored in the `issuer` variable. | The issuer URI for the authorization server. It should look something like this: `https://example.okta.com/oauth2/default` | Your Auth0 tenant's **Domain** with a leading `https://` and trailing slash `/`, e.g., `https://<your-tenant>.<region>.auth0.com/` |
-| Client ID   | Your Azure application ID.        | The **Client ID** in your Okta application. | Your Auth0 application's **Client ID** value. |
-| Client Secret | Azure client secret.| The **Client Secret** in your Okta application. | Your Auth0 application's **Client Secret** value. |
-
-You can test your configuration by navigating to the Dev Portal and using your IdP credentials to log in. 
\ No newline at end of file
+* [Configure generic SSO for a Konnect Org](/konnect/org-management/sso/)
+* [IdP SSO attribute mapping reference](/konnect/reference/sso)
```
Comment (smritikjaggi) [inline app/konnect/dev-portal/access-and-approval/sso.md:20]: Should we add this section for Dev Portal "Configure Okta" page as well: https://docs.konghq.com/konnect/org-management/okta-idp/#add-a-user-to-your-application?

File: app/_includes/md/konnect/generic-sso.md (hunk 4 above)
Comment (cloudjumpercat) [reply app/_includes/md/konnect/generic-sso.md:52]: Sure, I'll add that!

File: app/_includes/md/konnect/generic-sso.md (hunk 5 above)
Comment (cloudjumpercat) [reply app/_includes/md/konnect/generic-sso.md:106]: I moved that section to it's own page: https://deploy-preview-8118--kongdocs.netlify.app/konnect/reference/sso/ (trying to separate reference content from how to content). I can add to that reference page if we have mappings that are specific to Dev Portal.

File: app/konnect/dev-portal/access-and-approval/sso.md (hunk 6 above)
Comment (cloudjumpercat) [reply app/konnect/dev-portal/access-and-approval/sso.md:20]: I can add a link to that reference page to the Okta page

File: app/_includes/md/konnect/generic-sso.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+    * **Claim Mappings**: Customize the mapping of required attributes to a different claim in the `id_token` {{site.konnect_short_name}} receives from the IdP. By default, {{site.konnect_short_name}} requires three attributes: Name, Email, and Groups. The values in these attributes are mapped as follows:
+        * `name`: Used as the {{site.konnect_short_name}} account's `full_name`.
+        * `email`: Used as the {{site.konnect_short_name}} account's `email`.
+        * `groups`: Used to map users to teams defined in the team mappings upon login.
+{% if include.desc == "Dev Portal" %}
+1. Optional: [Map existing developer teams from IdP groups to {{site.konnect_short_name}} Dev Portal teams](/konnect/dev-portal/access-and-approval/add-teams/).
+{% endif %}
+{% if include.desc == "Konnect Org" %}
+1. Optional: To map existing teams from IdP groups to {{site.konnect_short_name}}, do the following:
+    1. Configure group claims in your IdP application. Be sure to add the following to the scope: `openid`, `email`, `profile`
```
Comment (cloudjumpercat) [inline app/_includes/md/konnect/generic-sso.md:56]: I feel slightly unsure about this step. I needed to make it generic, and I think groups claim is the correct name.

File: app/_includes/md/konnect/okta-sso.md [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+{% navtabs %}
+{% navtab OIDC %}
+1. Create a [new OIDC application in Okta](https://help.okta.com/oie/en-us/content/topics/apps/apps_app_integration_wizard_oidc.htm) to manage {{site.konnect_saas}} account integration. Configure the following settings:
+    * **Application Type**: Web Application
+    * **Grant type**: Authorization Code
+    {% if include.desc == "Konnect Org" %}
+    * **Sign-in redirect URIs**: `https://cloud.konghq.com/login` (This is a placeholder value that you'll replace later)
+    * **Sign-out redirect URIs**: `https://cloud.konghq.com/login` (This is a placeholder value that you'll replace later)
+    {% endif %}
+    {% if include.desc == "Dev Portal" %}
```
Comment (cloudjumpercat) [inline app/_includes/md/konnect/okta-sso.md:23]: For reviewers: This if include.desc isn't working in the output. It's showing the Konnect Org content on both the Dev Portal pages and Konnect Org pages. Any idea what I'm doing wrong? I didn't get any errors in the Terminal when I ran this locally.

File: app/_data/docs_nav_konnect.yml [hunk 9]
```diff
@@ -183,8 +183,10 @@
           url: /dev-portal/access-and-approval/add-teams/
         - text: Manage Application Registrations
           url: /dev-portal/access-and-approval/manage-app-connections/
-        - text: Configure SSO for Dev Portal
+        - text: Configure generic SSO for Dev Portal
           url: /dev-portal/access-and-approval/sso/
+        - text: Configure Okta SSO
```
Comment (lmilan) [inline app/_data/docs_nav_konnect.yml:188]: Should Dev Portal be mentioned in this title too?

**Review by lmilan — APPROVED** [review]
I just left one question, otherwise it looks good to me!

File: app/_data/docs_nav_konnect.yml (hunk 9 above)
Comment (cloudjumpercat) [reply app/_data/docs_nav_konnect.yml:188]: Good catch! I'll add that

Comment (cloudjumpercat) [reply app/_data/docs_nav_konnect.yml:188]:
```suggestion
        - text: Configure Okta SSO for Dev Portal
```

**Pull request merged**